wgsl/index.bs

```diff
@@ -1366,86 +1375,360 @@ Therefore, a data value must not be placed in the padding at the end of a struct
+        See [[#array-access-expr]].
+    * Given a reference with a structure store type, appending a member access phrase
+        results in a reference to the named member of the structure.
+        See [[#struct-access-expr]].
+
+<div class='example wgsl' heading='Component reference from a composite value reference'>
+  <xmp highlight='rust'>
+    struct S {
+        age: i32;
+        weight: f32;
+    };
+    var<private> person : S;
+
+    fn f() -> void {
+        var uv: vec2<f32>;
+        // Evaluate the left-hand side of the assignment:
+        //   Evaluate 'uv.x' to yield a reference:
+        //   1. First evaluate 'uv', yielding a reference to the storage for
+        //      the 'uv' variable. The result has type ref<function,vec2<f32>>.
+        //   2. Then apply the '.x' vector access phrase, yielding a reference to
+        //      the storage for the first component of the vector pointed at by the
+        //      reference value from the previous step.
+        //      The result has type ref<function,f32>.
+        // Evaluating the right-hand side of the assignment yields the f32 value 1.0.
+        // Store the f32 value 1.0 into the storage memory locations referenced by uv.x.
+        uv.x = 1.0;
+
+        // Evaluate the left-hand side of the assignment:
+        //   Evaluate 'uv[1]' to yield a reference:
+        //   1. First evaluate 'uv', yielding a reference to the storage for
+        //      the 'uv' variable. The result has type ref<function,vec2<f32>>.
+        //   2. Then apply the '[1]' array index phrase, yielding a reference to
+        //      the storage for second component of the vector referenced from
+        //      the previous step.  The result has type ref<function,f32>.
+        // Evaluating the right-hand side of the assignment yields the f32 value 2.0.
+        // Store the f32 value 2.0 into the storage memory locations referenced by uv[1].
+        uv[1] = 2.0;
+
+        var m: mat3x2<f32>;
+        // When evaluating 'm[2]':
+        // 1. First evaluate 'm', yielding a reference to the storage for
+        //    the 'm' variable. The result has type ref<function,mat3x2<f32>>.
+        // 2. Then apply the '[2]' array index phrase, yielding a reference to
+        //    the storage for the third column vector pointed at by the reference
+        //    value from the previous step.
+        //    Therefore the 'm[2]' expression has type ref<function,vec2<f32>>.
+        // The 'const' declaration is for type vec2<f32>, so the declaration
+        // statement requires the initializer to be of type vec2<f32>.
+        // The Load Rule applies (because no other type rule can apply), and
+        // the evaluation of the initializer yields the vec2<f32> value loaded
+        // from the memory locations referenced by m[2] at the time the declaration
+        // is executed.
+        const p_m_col2: vec2<f32> = m[2];
+
+        var A: array<i32,5>;
```

Comment:
This and the next example should be moved into the pointers example section.

Reply:
The main way to make a pointer is &.
So I'll rewrite these to avoid &, and use the Load Rule. I'll do that instead of moving these down.